In [1]:
import nltk 
nltk.download()

showing info https://raw.githubusercontent.com/nltk/nltk_data/gh-pages/index.xml


True

In [11]:
from nltk.tokenize import sent_tokenize
text="""Dogs ,Canis lupus familiaris, are domesticated mammals, not natural wild animals. They were originally bred from
wolves. They have been bred by humans for a long time, and were the first animals ever to be domesticated.
Today, some dogs are used as pets, others are used to help humans do their work. They are a popular pet because they are usually playful, friendly, loyal and listen to humans. Thirty million dogs in the United States are registered as pets. Dogs eat both meat and vegetables, often mixed together and sold in stores as dog food. Dogs often have jobs, including as police dogs, army dogs, assistance dogs, fire dogs, messenger dogs, hunting dogs, herding dogs, or rescue dogs.
They are sometimes called "canines" from the Latin word for dog - canis. Sometimes people also use "dog" to describe other
canids, such as wolves. A baby dog is called a pup or puppy. A dog is called a puppy until it is about one year old.
Dogs are sometimes referred to as "man’s best friend" because they are kept as domestic pets and are usually loyal and like being around humans."""

In [21]:
tokenized_text=sent_tokenize(text) 
print(tokenized_text)
print(tokenized_text[2])

['Dogs ,Canis lupus familiaris, are domesticated mammals, not natural wild animals.', 'They were originally bred from\nwolves.', 'They have been bred by humans for a long time, and were the first animals ever to be domesticated.', 'Today, some dogs are used as pets, others are used to help humans do their work.', 'They are a popular pet because they are usually playful, friendly, loyal and listen to humans.', 'Thirty million dogs in the United States are registered as pets.', 'Dogs eat both meat and vegetables, often mixed together and sold in stores as dog food.', 'Dogs often have jobs, including as police dogs, army dogs, assistance dogs, fire dogs, messenger dogs, hunting dogs, herding dogs, or rescue dogs.', 'They are sometimes called "canines" from the Latin word for dog - canis.', 'Sometimes people also use "dog" to describe other\ncanids, such as wolves.', 'A baby dog is called a pup or puppy.', 'A dog is called a puppy until it is about one year old.', 'Dogs are sometimes refer

How many sentences in the paragraph?\
13 sentences

What is the third sentences in the paragraph?\
They have been bred by humans for a long time, and were the first animals ever to be domesticated.

In [25]:
from nltk.tokenize import word_tokenize 
tokenized_word=word_tokenize(text) 
print(tokenized_word)
tokenized_word_0 = word_tokenize(tokenized_text[0])
print(len(tokenized_word_0))
print(tokenized_word)
tokenized_word_13 = word_tokenize(tokenized_text[12])
print(len(tokenized_word_13))

['Dogs', ',', 'Canis', 'lupus', 'familiaris', ',', 'are', 'domesticated', 'mammals', ',', 'not', 'natural', 'wild', 'animals', '.', 'They', 'were', 'originally', 'bred', 'from', 'wolves', '.', 'They', 'have', 'been', 'bred', 'by', 'humans', 'for', 'a', 'long', 'time', ',', 'and', 'were', 'the', 'first', 'animals', 'ever', 'to', 'be', 'domesticated', '.', 'Today', ',', 'some', 'dogs', 'are', 'used', 'as', 'pets', ',', 'others', 'are', 'used', 'to', 'help', 'humans', 'do', 'their', 'work', '.', 'They', 'are', 'a', 'popular', 'pet', 'because', 'they', 'are', 'usually', 'playful', ',', 'friendly', ',', 'loyal', 'and', 'listen', 'to', 'humans', '.', 'Thirty', 'million', 'dogs', 'in', 'the', 'United', 'States', 'are', 'registered', 'as', 'pets', '.', 'Dogs', 'eat', 'both', 'meat', 'and', 'vegetables', ',', 'often', 'mixed', 'together', 'and', 'sold', 'in', 'stores', 'as', 'dog', 'food', '.', 'Dogs', 'often', 'have', 'jobs', ',', 'including', 'as', 'police', 'dogs', ',', 'army', 'dogs', ',', 

How many words in the paragraph?\
229 words

How many words in the first sentence?\
15 words

How many words in the last sentence?\
30 words

Do you find repeat words?\
yes

In [26]:
from nltk.probability import FreqDist 
fdist = FreqDist(tokenized_word) 
print(fdist)

<FreqDist with 120 samples and 229 outcomes>


What is the output from print(fdist)?\
120 samples don't count a repeated word.\
229 outcomes are count all of output.

In [27]:
fdist

FreqDist({',': 18, '.': 13, 'are': 10, 'dogs': 10, 'as': 7, 'and': 6, 'to': 5, 'dog': 5, 'Dogs': 4, 'They': 4, ...})

What is the top ten most common word?

In [30]:
fdist.most_common(10)

[(',', 18),
 ('.', 13),
 ('are', 10),
 ('dogs', 10),
 ('as', 7),
 ('and', 6),
 ('to', 5),
 ('dog', 5),
 ('Dogs', 4),
 ('They', 4)]

In [31]:
my_stopwords = ["is", "a", "this", ","]

In [32]:
from nltk.corpus import stopwords 
nltk_stop_words=set(stopwords.words("english"))
print(nltk_stop_words)

{'him', 'hers', 'which', 'in', 'i', 'were', 'she', 'themselves', 'once', "don't", 'any', 'most', 'weren', 'those', 'out', 'ma', 'before', 'on', 'above', 'through', "should've", 'up', 'our', "that'll", 'between', 'because', 'you', 'against', 'herself', 'so', 'than', 'was', 'y', 'after', 'about', 'should', 'ain', 'd', 'm', 'theirs', 'her', 'll', 'then', 'again', 'isn', 'more', 'whom', 're', "won't", 'only', "isn't", 'these', 'o', 'own', 've', 'doing', 'too', 'wouldn', 'few', 'same', 'your', "you're", 'its', 'but', 'below', 'just', 'yourself', 'for', 'have', 'himself', 'into', 'me', 'myself', "you'll", 'itself', 'will', 'here', 'having', 'is', 'don', 'did', 'ourselves', 't', "she's", 'their', 'there', 'being', 'be', 'doesn', 'while', 'the', 'hadn', 'under', 'very', "shan't", 'been', 'at', 'them', 'now', 'am', 'won', 'yourselves', 'can', 'by', 'each', "haven't", 'didn', "hadn't", 'off', 'over', "hasn't", "doesn't", 'further', 'during', 'hasn', 'he', 'when', 'wasn', 'other', 'it', 'or', 'yo

What are those stopwords provided by NLTK?\
Remove the stopped word.

In [35]:
filtered_words=[]
for w in tokenized_word:
    if w not in my_stopwords: 
        filtered_words.append(w)
print("Tokenized Sentence:",tokenized_word) 
print("\n")
print("Filterd Sentence:",filtered_words)

Tokenized Sentence: ['Dogs', ',', 'Canis', 'lupus', 'familiaris', ',', 'are', 'domesticated', 'mammals', ',', 'not', 'natural', 'wild', 'animals', '.', 'They', 'were', 'originally', 'bred', 'from', 'wolves', '.', 'They', 'have', 'been', 'bred', 'by', 'humans', 'for', 'a', 'long', 'time', ',', 'and', 'were', 'the', 'first', 'animals', 'ever', 'to', 'be', 'domesticated', '.', 'Today', ',', 'some', 'dogs', 'are', 'used', 'as', 'pets', ',', 'others', 'are', 'used', 'to', 'help', 'humans', 'do', 'their', 'work', '.', 'They', 'are', 'a', 'popular', 'pet', 'because', 'they', 'are', 'usually', 'playful', ',', 'friendly', ',', 'loyal', 'and', 'listen', 'to', 'humans', '.', 'Thirty', 'million', 'dogs', 'in', 'the', 'United', 'States', 'are', 'registered', 'as', 'pets', '.', 'Dogs', 'eat', 'both', 'meat', 'and', 'vegetables', ',', 'often', 'mixed', 'together', 'and', 'sold', 'in', 'stores', 'as', 'dog', 'food', '.', 'Dogs', 'often', 'have', 'jobs', ',', 'including', 'as', 'police', 'dogs', ',', '

Can you apply the stopwords provided by NLTK (nltk stop words) to your text?\
yes

What is the output? Can you try some other words such as derivational, multiplying?

In [37]:
from nltk.stem import PorterStemmer 
stem = PorterStemmer()
word = "connection" 
stem.stem(word)

'connect'

What is the output?

In [39]:
from nltk.stem import PorterStemmer 
stem = PorterStemmer()
stemmed_words=[]
for w in filtered_words:
    stemmed_words.append(stem.stem(w))
print("Filtered Words:",filtered_words) 
print("\n")
print("Stemmed Words:",stemmed_words)

Filtered Words: ['Dogs', 'Canis', 'lupus', 'familiaris', 'are', 'domesticated', 'mammals', 'not', 'natural', 'wild', 'animals', '.', 'They', 'were', 'originally', 'bred', 'from', 'wolves', '.', 'They', 'have', 'been', 'bred', 'by', 'humans', 'for', 'long', 'time', 'and', 'were', 'the', 'first', 'animals', 'ever', 'to', 'be', 'domesticated', '.', 'Today', 'some', 'dogs', 'are', 'used', 'as', 'pets', 'others', 'are', 'used', 'to', 'help', 'humans', 'do', 'their', 'work', '.', 'They', 'are', 'popular', 'pet', 'because', 'they', 'are', 'usually', 'playful', 'friendly', 'loyal', 'and', 'listen', 'to', 'humans', '.', 'Thirty', 'million', 'dogs', 'in', 'the', 'United', 'States', 'are', 'registered', 'as', 'pets', '.', 'Dogs', 'eat', 'both', 'meat', 'and', 'vegetables', 'often', 'mixed', 'together', 'and', 'sold', 'in', 'stores', 'as', 'dog', 'food', '.', 'Dogs', 'often', 'have', 'jobs', 'including', 'as', 'police', 'dogs', 'army', 'dogs', 'assistance', 'dogs', 'fire', 'dogs', 'messenger', 'do

In [40]:
from nltk.stem.wordnet import WordNetLemmatizer 
lem = WordNetLemmatizer()
word = "connecting" 
lem.lemmatize(word, pos="v")

'connect'

In [41]:
from nltk.stem.wordnet import WordNetLemmatizer 
lem = WordNetLemmatizer()
word = "flying" 
print(lem.lemmatize(word, pos="v"))
print(lem.lemmatize(word, pos="n"))

fly
flying


What is the different between using v and n?\
fly is v\
flying is n

In [42]:
sent = "I am studying at ICT" 
tokens=nltk.word_tokenize(sent)
print(tokens) 
nltk.pos_tag(tokens)

['I', 'am', 'studying', 'at', 'ICT']


[('I', 'PRP'),
 ('am', 'VBP'),
 ('studying', 'VBG'),
 ('at', 'IN'),
 ('ICT', 'NNP')]

Can you try to interpret the result?\
PRP is personal pronoun\
VBP is verb, single present\
VBG is verb, gerund/present participle taking\
IN is preposition, subordination conjunction\
NNP is proper noun